flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsMemoryDataManager.java

```diff
@@ -72,8 +73,8 @@ public class HsMemoryDataManager implements HsSpillingInfoProvider, HsMemoryData
 
     private final AtomicInteger numUnSpillBuffers = new AtomicInteger(0);
 
-    private final Map<Integer, HsSubpartitionViewInternalOperations> subpartitionViewOperationsMap =
-            new ConcurrentHashMap<>();
+    private final List<Map<Integer, HsSubpartitionViewInternalOperations>>
+            subpartitionViewOperationsMap;
```

Comment:
Definition should be documented.

Reply:
Can we replace the outer `List` with an array here, to align with `subpartitionMemoryDataManagers`?

Reply:
Why the inner is a `Map` rather than a `List`. With a list, we won't need the `consumerIdCounter` in `HsResultPartition`.

Reply:
I didn't change the List into an array because I didn't want to introduce a raw type Map.

Reply:
Document added.